zookeeper-server/src/main/java/org/apache/zookeeper/server/ZooKeeperServer.java

```diff
@@ -546,7 +554,9 @@ public void loadData() throws IOException, InterruptedException {
                         .forEach(session -> killSession(session, zkDb.getDataTreeLastProcessedZxid()));
 
         // Make a clean snapshot
```

Comment:
Please update the comment in what cases needSnapshot would be true/false and why.

Reply:
Sounds good, added a comment 👍